pinot-ingestion-jobs/pinot-ingestion-common/src/main/java/org/apache/pinot/ingestion/common/DefaultControllerRestApi.java

```diff
@@ -121,7 +122,47 @@ public void pushSegments(FileSystem fileSystem, List<Path> tarFilePaths) {
             LOGGER.error("Caught exception while pushing segment: {} to location: {}, retry {}/{}", segmentName,
                 pushLocation, retry, _retry, e);
             if (retry == _retry) {
-              throw new RuntimeException(String.format("Failed to push segment %s to %s with %d retries",segmentName, pushLocation, retry), e);
+              throw new RuntimeException(
+                  String.format("Failed to push segment %s to %s with %d retries", segmentName, pushLocation, retry),
+                  e);
+            }
+            try {
+              // Exponential back-off, max sleep time is 64 seconds.
+              Thread.sleep(1000 * (int) Math.pow(2, Math.min(retry + 1, 6)));
+            } catch (InterruptedException ex) {
+              // Swallow
+            }
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void pushSegments(PinotFS fileSystem, List<String> tarFilePaths) {
```

Comment:
Seems like there's a lot of duplicate code between this version and the previous version of this method. Would be good to refactor so that we avoid duplicates.

Reply:
I missed this. @fx19880617, can we not depend on pinot-ingestion-common? My preference is to completely deprecate the existing pinot-ingestion-common and pinot-hadoop.

Reply:
+1

Reply:
I will remove the dependencies from `pinot-ingestion-common`